src/frontend/Komponenter/Behandling/VedtakOgBeregning/Barnetilsyn/revurderFraUtils.ts

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from 'uuid';
+import { IUtgiftsperiode, IvedtakForBarnetilsyn } from '../../../../App/typer/vedtak';
+import { månedÅrTilDate, plusMåneder, tilÅrMåned } from '../../../../App/utils/dato';
+
+const lagMidlertidigOpphør = (fra: string, til: string): IUtgiftsperiode => ({
+    årMånedFra: fra,
+    årMånedTil: til,
+    barn: [],
+    utgifter: undefined,
+    erMidlertidigOpphør: true,
+    endretKey: uuidv4(),
+});
+/**
+ * Fyller hull mellom 2 perioder med midlertidig opphør
+ * Sjekker om forrige sin tildato plus en måned er den samme måneden som neste periode sin fra-måned
+ */
+export const fyllHullMedOpphør = (
+    acc: IUtgiftsperiode[],
+    periode: IUtgiftsperiode
+): IUtgiftsperiode[] => {
+    const forrige = acc[acc.length - 1];
+    if (forrige && forrige.årMånedTil && periode.årMånedFra) {
+        const hullFra = tilÅrMåned(plusMåneder(månedÅrTilDate(forrige.årMånedTil), 1));
+        const hullTil = tilÅrMåned(plusMåneder(månedÅrTilDate(periode.årMånedFra), -1));
+        if (hullFra !== periode.årMånedFra) {
+            acc.push(lagMidlertidigOpphør(hullFra, hullTil));
+        }
+    }
+    acc.push(periode);
+    return acc;
+};
+
+/**
+ * Lager en periode som legges før tidligere perioder hvis revurderes fra er før tidligere dato
+ */
+export const revurderFraInitPeriode = <T>(
+    vedtakshistorikk: IvedtakForBarnetilsyn,
+    revurderesFra: string,
+    periode: (revurderesFra: string) => T
```

Comment:
Denne ble litt magisk - er det alltid tomperiode her, egentlig? 🤔 Kunne denne droppet å ta inn tomperioden og heller kalle på den direkte?

Reply:
Ja, kan vel det for barnetilsyn :) Dette er liknende kode som for OS

Reply:
[f3be66b](https://github.com/navikt/familie-ef-sak-frontend/pull/1863/commits/f3be66bb45feee2009525f91be09ee337411a0cc)